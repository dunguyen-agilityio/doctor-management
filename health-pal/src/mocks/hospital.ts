import { Hospital } from '@app/models/hospital'

export const MOCK_HOSPITALS: Hospital[] = [
  {
    id: 1,
    documentId: '123e4567-e89b-12d3-a456-426614174000',
    name: 'Sunrise Hospital',
    description: 'A premier healthcare facility offering top-tier services in Springfield.',
    address: '123 Main St, Springfield, USA',
    image: { id: 1, url: 'https://picsum.photos/200', alt: 'Sunrise Hospital' },
    rating: 4.2,
    doctors: [],
    reivewCouter: 300,
    reviews: [],
    type: 'Clinic',
  },
  {
    id: 2,
    documentId: '987fcdeb-12ab-34cd-5678-901234567890',
    name: 'Mercy Hospital',
    description: 'Leading medical center in Metropolis.',
    address: '456 Oak Ave, Metropolis, Canada',
    image: { id: 2, url: 'https://picsum.photos/201', alt: 'Mercy Hospital' },
    doctors: [],
    reivewCouter: 300,
    reviews: [],
    rating: 1,
    type: 'Hospital',
  },
]
