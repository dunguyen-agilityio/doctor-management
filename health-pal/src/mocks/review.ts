import { Review } from '@/models/review'

export const MOCK_REVIEWS = [
  {
    name: 'Emily Anderson',
    rating: 5,
    comment:
      'Dr. Patel is a true professional who genuinely cares about his patients. I highly recommend Dr. Patel to',
    documentId: 'doc-1',
    id: 1,
  },
  {
    name: 'Emily Anderson',
    rating: 5,
    comment:
      'Dr. Patel is a true professional who genuinely cares about his patients. I highly recommend Dr. Patel to',
    image: require('@assets/images/user01.png'),
    documentId: 'doc-2',
    id: 2,
  },
  {
    name: 'Emily Anderson',
    rating: 5,
    comment:
      'Dr. Patel is a true professional who genuinely cares about his patients. I highly recommend Dr. Patel to',
    image: require('@assets/images/user01.png'),
    documentId: 'doc-3',
    id: 3,
  },
  {
    name: 'Emily Anderson',
    rating: 5,
    comment:
      'Dr. Patel is a true professional who genuinely cares about his patients. I highly recommend Dr. Patel to',
    image: require('@assets/images/user01.png'),
    documentId: 'doc-4',
    id: 4,
  },
] as Review[]
