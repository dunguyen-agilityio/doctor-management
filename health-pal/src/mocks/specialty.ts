import { Specialty } from '@/models/specialty'

export const MOCK_SPECIALTY: Specialty = {
  id: 1,
  documentId: '123e4567-e89b-12d3-a456-426614174000',
  name: 'Cardiology',
  description: 'Specialized in heart-related conditions and treatments.',
  doctors: [],
}
