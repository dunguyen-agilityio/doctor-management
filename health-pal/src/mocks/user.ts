import { User } from '@/models/user'

export const MOCK_USER: User = {
  id: 1,
  name: 'John Doe',
  nickname: 'jdoe',
  email: 'jdoe@example.com',
  documentId: '123e4567-e89b-12d3-a456-426614174000',
  dateOfBirth: new Date('1990-01-01'),
  gender: true,
  avatar: {
    id: 1,
    url: 'https://example.com/avatar.jpg',
  },
  username: 'jdoe',
  bio: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
}
