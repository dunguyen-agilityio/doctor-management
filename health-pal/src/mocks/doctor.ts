import { TDoctorData } from '@/models/doctor'

import { TDoctorBanner } from '@/types/doctor'

import { MOCK_HOSPITALS } from './hospital'
import { MOCK_REVIEWS } from './review'
import { MOCK_SPECIALTY } from './specialty'
import { MOCK_USER } from './user'

export const MOCK_DOCTORS: TDoctorData[] = [
  {
    ...MOCK_USER,
    documentId: 'doc_001',
    id: 1,
    name: 'Dr. John Smith',
    reviewCounter: 5,
    rating: 4.8,
    clinic: MOCK_HOSPITALS[0],
    dateOfBirth: new Date('1980-01-01'),
    email: 'john.smith@example.com',
    specialty: MOCK_SPECIALTY,
    avatar: {
      id: 1,
      url: 'https://picsum.photos/id/91/3504/2336',
    },
    reviews: MOCK_REVIEWS,
    favoriteId: 'fav_001',
    users_permissions_user: MOCK_USER,
  },
]

export const DOCTOR_BANNER: TDoctorBanner[] = [...new Array(6).keys()].map(() => ({
  title: `Looking for\nSpecialist Doctors?`,
  description: `Schedule an appointment with\nour top doctors.`,
}))
