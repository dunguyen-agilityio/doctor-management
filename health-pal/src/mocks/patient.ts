import { Patient } from '@/models/patient'

import { MOCK_USER } from './user'

export const MOCK_PATIENT: Patient = {
  bookings: [],
  ...MOCK_USER,
}
