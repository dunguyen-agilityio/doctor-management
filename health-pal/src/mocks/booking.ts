import { BookingData } from '@/models/booking'

import { BOOKING_TABS } from '@/types/booking'

import { MOCK_DOCTORS } from './doctor'
import { MOCK_PATIENT } from './patient'

export const MOCK_BOOKING: BookingData = {
  id: 1,
  documentId: 'doc-1',
  date: '2025-06-01',
  time: '09:00 AM',
  type: BOOKING_TABS.UPCOMING,
  patient: MOCK_PATIENT,
  doctor: MOCK_DOCTORS[0],
}
