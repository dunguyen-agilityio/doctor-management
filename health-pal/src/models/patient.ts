import { Booking } from './booking'
import { User } from './user'

export class Patient extends User {
  bookings: Booking[]

  constructor(patient: Patient) {
    super(patient)
    this.bookings = patient.bookings
  }
}
