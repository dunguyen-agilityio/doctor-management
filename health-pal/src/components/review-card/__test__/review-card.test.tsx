import { MOCK_PATIENT } from '@/mocks/patient'
import { render } from '@utils-test'

import ReviewCard from '..'

// Adjust path

describe('ReviewCard', () => {
  const reviewData = {
    name: 'John Doe',
    rating: 4.5,
    comment: 'Great experience!',
    image: 'https://example.com/avatar.jpg',
    patient: MOCK_PATIENT,
    id: 1,
    documentId: 'doc-1',
  }

  it('matches snapshot', () => {
    const { toJSON } = render(<ReviewCard {...reviewData} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
