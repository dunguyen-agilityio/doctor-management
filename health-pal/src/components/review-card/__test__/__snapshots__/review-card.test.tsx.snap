// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewCard matches snapshot 1`] = `
[
  <View
    aria-label="Notifications (F8)"
    pointerEvents="box-none"
    role="region"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100000,
      }
    }
    tabIndex={-1}
  >
    <View
      focusable={false}
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "column",
          "maxWidth": "100%",
          "position": "absolute",
        }
      }
    />
  </View>,
  <View
    style={
      {
        "backgroundColor": "white",
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "white",
          "flexDirection": "column",
          "gap": 12,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "flexDirection": "row",
            "gap": 10,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "red",
              "borderRadius": 56,
              "height": 56,
              "overflow": "hidden",
              "width": 56,
            }
          }
        >
          <ViewManagerAdapter_ExpoImage
            cachePolicy="disk"
            containerViewRef={"[React.ref]"}
            contentFit="cover"
            contentPosition={
              {
                "left": "50%",
                "top": "50%",
              }
            }
            height="100%"
            nativeViewRef={"[React.ref]"}
            onError={[Function]}
            onLoad={[Function]}
            onLoadEnd={[Function]}
            onLoadStart={[Function]}
            onProgress={[Function]}
            placeholder={
              [
                {
                  "height": 16,
                  "uri": "blurhash:/LDPzx.00%5B93GrBXnp0McR5X9NHn%23",
                  "width": 16,
                },
              ]
            }
            source={
              [
                {
                  "uri": "https://example.com/avatar.jpg",
                },
              ]
            }
            style={
              {
                "height": "100%",
                "width": "100%",
              }
            }
            transition={null}
            width="100%"
          />
          <View
            style={
              {
                "flexDirection": "column",
                "left": "50%",
                "position": "absolute",
                "top": "50%",
                "transform": [
                  {
                    "translateX": -12,
                  },
                  {
                    "translateY": -12,
                  },
                ],
              }
            }
          >
            <ActivityIndicator
              color="#1c64f2"
              size="small"
            />
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "white",
              "flexDirection": "column",
              "gap": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "#1c2a3a",
                "fontFamily": "Inter_700Bold",
                "fontSize": 18,
                "lineHeight": 27,
              }
            }
            suppressHighlighting={true}
          >
            John Doe
          </Text>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "gap": 8,
              }
            }
            testID="stars"
          >
            <Text
              style={
                {
                  "color": "#1c2a3a",
                  "fontFamily": "Inter_600SemiBold",
                  "fontSize": 12,
                  "lineHeight": 18,
                }
              }
              suppressHighlighting={true}
              testID="stars-text"
            >
              4.5
            </Text>
            <View
              style={[]}
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={12}
                bbWidth={12}
                fill="none"
                focusable={false}
                height={12}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": 12,
                      "width": 12,
                    },
                  ]
                }
                testID="star-0"
                vbHeight={21}
                vbWidth={21}
                width={12}
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    clipRule={0}
                    d="M10.6136 0.818184C9.9994 0.849607 9.39008 1.33065 8.91935 2.25989L7.1322 5.83418C6.89138 6.3285 6.24497 6.81014 5.69995 6.89887L2.46787 7.43121C0.401877 7.7861 -0.0797662 9.28173 1.40319 10.7647L3.92548 13.287C4.34374 13.7052 4.58457 14.5291 4.47049 15.1375L3.74803 18.2555C3.17766 20.7144 4.49584 21.665 6.66324 20.3849L9.69252 18.585C9.94502 18.4324 10.2764 18.3504 10.6136 18.3392V0.818184Z"
                    fill={
                      {
                        "payload": 4294956800,
                        "type": 0,
                      }
                    }
                    fillRule={0}
                    propList={
                      [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                  <RNSVGPath
                    clipRule={0}
                    d="M10.6133 18.3409C11.0039 18.328 11.4023 18.4099 11.6948 18.5868L14.7241 20.3866C16.8915 21.6794 18.2096 20.7161 17.6393 18.2572L16.9168 15.1392C16.7774 14.5435 17.0182 13.7196 17.4365 13.3014L19.9588 10.7791C21.4417 9.28345 20.9601 7.78782 18.8941 7.4456L15.662 6.91326C15.117 6.81186 14.4706 6.34289 14.2297 5.84857L12.4426 2.27428C11.9374 1.27039 11.2725 0.786183 10.6133 0.819899V18.3409Z"
                    fill={
                      {
                        "payload": 4294956800,
                        "type": 0,
                      }
                    }
                    fillRule={0}
                    propList={
                      [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={12}
                bbWidth={12}
                fill="none"
                focusable={false}
                height={12}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": 12,
                      "width": 12,
                    },
                  ]
                }
                testID="star-1"
                vbHeight={21}
                vbWidth={21}
                width={12}
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    clipRule={0}
                    d="M10.6136 0.818184C9.9994 0.849607 9.39008 1.33065 8.91935 2.25989L7.1322 5.83418C6.89138 6.3285 6.24497 6.81014 5.69995 6.89887L2.46787 7.43121C0.401877 7.7861 -0.0797662 9.28173 1.40319 10.7647L3.92548 13.287C4.34374 13.7052 4.58457 14.5291 4.47049 15.1375L3.74803 18.2555C3.17766 20.7144 4.49584 21.665 6.66324 20.3849L9.69252 18.585C9.94502 18.4324 10.2764 18.3504 10.6136 18.3392V0.818184Z"
                    fill={
                      {
                        "payload": 4294956800,
                        "type": 0,
                      }
                    }
                    fillRule={0}
                    propList={
                      [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                  <RNSVGPath
                    clipRule={0}
                    d="M10.6133 18.3409C11.0039 18.328 11.4023 18.4099 11.6948 18.5868L14.7241 20.3866C16.8915 21.6794 18.2096 20.7161 17.6393 18.2572L16.9168 15.1392C16.7774 14.5435 17.0182 13.7196 17.4365 13.3014L19.9588 10.7791C21.4417 9.28345 20.9601 7.78782 18.8941 7.4456L15.662 6.91326C15.117 6.81186 14.4706 6.34289 14.2297 5.84857L12.4426 2.27428C11.9374 1.27039 11.2725 0.786183 10.6133 0.819899V18.3409Z"
                    fill={
                      {
                        "payload": 4294956800,
                        "type": 0,
                      }
                    }
                    fillRule={0}
                    propList={
                      [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={12}
                bbWidth={12}
                fill="none"
                focusable={false}
                height={12}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": 12,
                      "width": 12,
                    },
                  ]
                }
                testID="star-2"
                vbHeight={21}
                vbWidth={21}
                width={12}
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    clipRule={0}
                    d="M10.6136 0.818184C9.9994 0.849607 9.39008 1.33065 8.91935 2.25989L7.1322 5.83418C6.89138 6.3285 6.24497 6.81014 5.69995 6.89887L2.46787 7.43121C0.401877 7.7861 -0.0797662 9.28173 1.40319 10.7647L3.92548 13.287C4.34374 13.7052 4.58457 14.5291 4.47049 15.1375L3.74803 18.2555C3.17766 20.7144 4.49584 21.665 6.66324 20.3849L9.69252 18.585C9.94502 18.4324 10.2764 18.3504 10.6136 18.3392V0.818184Z"
                    fill={
                      {
                        "payload": 4294956800,
                        "type": 0,
                      }
                    }
                    fillRule={0}
                    propList={
                      [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                  <RNSVGPath
                    clipRule={0}
                    d="M10.6133 18.3409C11.0039 18.328 11.4023 18.4099 11.6948 18.5868L14.7241 20.3866C16.8915 21.6794 18.2096 20.7161 17.6393 18.2572L16.9168 15.1392C16.7774 14.5435 17.0182 13.7196 17.4365 13.3014L19.9588 10.7791C21.4417 9.28345 20.9601 7.78782 18.8941 7.4456L15.662 6.91326C15.117 6.81186 14.4706 6.34289 14.2297 5.84857L12.4426 2.27428C11.9374 1.27039 11.2725 0.786183 10.6133 0.819899V18.3409Z"
                    fill={
                      {
                        "payload": 4294956800,
                        "type": 0,
                      }
                    }
                    fillRule={0}
                    propList={
                      [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={12}
                bbWidth={12}
                fill="none"
                focusable={false}
                height={12}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": 12,
                      "width": 12,
                    },
                  ]
                }
                testID="star-3"
                vbHeight={21}
                vbWidth={21}
                width={12}
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    clipRule={0}
                    d="M10.6136 0.818184C9.9994 0.849607 9.39008 1.33065 8.91935 2.25989L7.1322 5.83418C6.89138 6.3285 6.24497 6.81014 5.69995 6.89887L2.46787 7.43121C0.401877 7.7861 -0.0797662 9.28173 1.40319 10.7647L3.92548 13.287C4.34374 13.7052 4.58457 14.5291 4.47049 15.1375L3.74803 18.2555C3.17766 20.7144 4.49584 21.665 6.66324 20.3849L9.69252 18.585C9.94502 18.4324 10.2764 18.3504 10.6136 18.3392V0.818184Z"
                    fill={
                      {
                        "payload": 4294956800,
                        "type": 0,
                      }
                    }
                    fillRule={0}
                    propList={
                      [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                  <RNSVGPath
                    clipRule={0}
                    d="M10.6133 18.3409C11.0039 18.328 11.4023 18.4099 11.6948 18.5868L14.7241 20.3866C16.8915 21.6794 18.2096 20.7161 17.6393 18.2572L16.9168 15.1392C16.7774 14.5435 17.0182 13.7196 17.4365 13.3014L19.9588 10.7791C21.4417 9.28345 20.9601 7.78782 18.8941 7.4456L15.662 6.91326C15.117 6.81186 14.4706 6.34289 14.2297 5.84857L12.4426 2.27428C11.9374 1.27039 11.2725 0.786183 10.6133 0.819899V18.3409Z"
                    fill={
                      {
                        "payload": 4294956800,
                        "type": 0,
                      }
                    }
                    fillRule={0}
                    propList={
                      [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          {
            "color": "#6b7280",
            "fontFamily": "Inter_400Regular",
            "fontSize": 14,
            "lineHeight": 21,
            "marginTop": 5,
          }
        }
        suppressHighlighting={true}
      >
        Great experience!
      </Text>
    </View>
  </View>,
]
`;
